content/articles/servlet-request-and-servlet-collaboration/SevletRequest and ServletCallaboration.md

```diff
@@ -0,0 +1,277 @@
+ <input type="submit" value="submit">
+</form>
+</body>
+</html>
+```
+*Controller.java servlet1 page*
+```java
+package com.lodoctor;
+
+import java.io.IOException;
+
+import jakarta.servlet.RequestDispatcher;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+
+
+public class Controller extends HttpServlet 
+{
+	
+	protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException 
+	{
+		response.setContentType("Text/html");
+		String select=request.getParameter("selectOption");
+		response.getWriter().append("<html><body style='background-color: red;'></body></html>");
+		if(select.equals("Bellow"))
+		{
+			RequestDispatcher rd=request.getRequestDispatcher("Under18Page");
+			rd.forward(request,response);
+		}
+		if(select.equals("Older"))
+		{
+			RequestDispatcher rd=request.getRequestDispatcher("Above18");
+			rd.include(request,response);
+		}
+		else
+		{
+			RequestDispatcher rd=request.getRequestDispatcher("htmlPage.html");
+			rd.forward(request,response);
+		}
+		
+	}
+
+	
+
+}
+
+```
+*Under18page.java servlet2 page*
+```java
+package com.lodoctor;
+
+import java.io.IOException;
+
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+public class Under18Page extends HttpServlet {
+	
+	protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException
+	{
+		response.setContentType("Text/html");
+		response.getWriter().append("<html><body style='background-color: red; color: white; text-align: center; padding :200px'><h1>you are under 18</h1></body></html>");
+	}
+
+
+}
+
+```
+*Above18.java servlet3 page*
+```java
+package com.lodoctor;
+
+import java.io.IOException;
+
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+public class Above18 extends HttpServlet {
+	private static final long serialVersionUID = 1L;
+
+   
+	protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException 
+	{
+		response.setContentType("Text/html");
+		response.getWriter().append("<html><body style='background-color: red; color: white; text-align: center; padding :200px'><h1>you are above 18</h1></body></html>");
+	}
+
+
+}
+
+```
+[Get project source code here]
+(https://drive.google.com/drive/folders/1S0S2SJKG1PIwRGX-zll25Y3mTrCOLy5D?usp=sharing)
+### Conclusion
+In conclusion, we learned some of the ServletRequest methods and ServletDispatcher methods and how to implement them with the help of examples. We discussed both these interfaces (ServletRequest and ServletDispatcher) and what they comprise.

```

Comment:
Highlight ServletRequest and ServletDispatcher

Reply:
I'm done sir.